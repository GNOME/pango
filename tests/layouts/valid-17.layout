{
  "context" : {
    "font" : "serif 12",
    "language" : "en-us",
    "base-gravity" : "south",
    "gravity-hint" : "natural",
    "base-dir" : "weak-ltr",
    "round-glyph-positions" : true,
    "transform" : [
      1,
      0,
      0,
      1,
      0,
      0
    ]
  },
  "comment" : "Hyphenation point should get replaced with a hyphen",
  "text" : "some‧bla‧bla‧bla",
  "font" : "Cantarell 11",
  "justify" : true,
  "width" : 102400,
  "output" : {
    "is-wrapped" : true,
    "is-ellipsized" : false,
    "unknown-glyphs" : 0,
    "width" : 96976,
    "height" : 38912,
    "log-attrs" : [
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "sentence-boundary" : true,
        "sentence-start" : true,
        "backspace-deletes-character" : true,
        "word-boundary" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-end" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true,
        "break-removes-preceding" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-end" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true,
        "break-removes-preceding" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "word-end" : true
      },
      {
        "line-break" : true,
        "char-break" : true,
        "cursor-position" : true,
        "word-start" : true,
        "backspace-deletes-character" : true,
        "break-inserts-hyphen" : true,
        "break-removes-preceding" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "char-break" : true,
        "cursor-position" : true,
        "break-inserts-hyphen" : true
      },
      {
        "line-break" : true,
        "mandatory-break" : true,
        "char-break" : true,
        "white" : true,
        "cursor-position" : true,
        "word-end" : true,
        "sentence-boundary" : true,
        "sentence-end" : true,
        "word-boundary" : true
      }
    ],
    "lines" : [
      {
        "start-index" : 0,
        "length" : 19,
        "paragraph-start" : true,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 0,
            "length" : 4,
            "text" : "some",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 417,
                "width" : 7326,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 370,
                "width" : 8508,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 358,
                "width" : 13628,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 2
              },
              {
                "glyph" : 287,
                "width" : 8508,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 3
              }
            ]
          },
          {
            "offset" : 4,
            "length" : 3,
            "text" : "‧",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "DejaVu Sans 11",
              "checksum" : "69ccd07023a72ceb27a5e5c22f728627353b60a198170f5e58dd7014221abf01",
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 2826,
                "width" : 5436,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 7,
            "length" : 3,
            "text" : "bla",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 272,
                "width" : 8508,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 349,
                "width" : 4412,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 244,
                "width" : 8508,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 2
              }
            ]
          },
          {
            "offset" : 10,
            "length" : 3,
            "text" : "‧",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "DejaVu Sans 11",
              "checksum" : "69ccd07023a72ceb27a5e5c22f728627353b60a198170f5e58dd7014221abf01",
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 2826,
                "width" : 5436,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          },
          {
            "offset" : 13,
            "length" : 3,
            "text" : "bla",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 272,
                "width" : 8508,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 349,
                "width" : 4412,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 244,
                "width" : 8508,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 2
              }
            ]
          },
          {
            "offset" : 16,
            "length" : 3,
            "text" : "‧",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "DejaVu Sans 11",
              "checksum" : "69ccd07023a72ceb27a5e5c22f728627353b60a198170f5e58dd7014221abf01",
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 4,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 2803,
                "width" : 5278,
                "x-offset" : 158,
                "is-cluster-start" : true,
                "log-cluster" : 0
              }
            ]
          }
        ]
      },
      {
        "start-index" : 19,
        "length" : 3,
        "paragraph-start" : false,
        "direction" : "ltr",
        "runs" : [
          {
            "offset" : 19,
            "length" : 3,
            "text" : "bla",
            "bidi-level" : 0,
            "gravity" : "south",
            "language" : "en-us",
            "script" : "latin",
            "font" : {
              "description" : "Cantarell 11",
              "checksum" : "5bcb6ee14ee9d210b2e91d643de1fe456e9d1aea770983fdb05951545efebbe2",
              "variations" : {
                "wght" : 0
              },
              "matrix" : [
                1,
                -0,
                -0,
                1,
                0,
                0
              ]
            },
            "flags" : 0,
            "y-offset" : 0,
            "start-x-offset" : 0,
            "end-x-offset" : 0,
            "glyphs" : [
              {
                "glyph" : 272,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 0
              },
              {
                "glyph" : 349,
                "width" : 4096,
                "is-cluster-start" : true,
                "log-cluster" : 1
              },
              {
                "glyph" : 244,
                "width" : 8192,
                "is-cluster-start" : true,
                "log-cluster" : 2
              }
            ]
          }
        ]
      }
    ]
  }
}
