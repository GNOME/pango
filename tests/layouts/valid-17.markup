justify=true,width=100
some|bla|bla|bla
