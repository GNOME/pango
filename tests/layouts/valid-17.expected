some|bla|bla|bla

--- parameters

wrapped: 1
ellipsized: 0
lines: 3
width: 102400

--- attributes

range 0 2147483647

--- directions

0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 

--- cursor positions

0(0) 1(0) 2(0) 3(0) 4(0) 5(0) 6(0) 7(0) 8(0) 9(0) 10(0) 11(0) 12(0) 12(1) 14(0) 15(0) 15(1) 17(0) 

--- lines

i=1, index=0, paragraph-start=1, dir=ltr 'some|bla|bla|'
i=2, index=13, paragraph-start=0, dir=ltr 'bla
'
i=3, index=17, paragraph-start=1, dir=ltr ''

--- runs

i=1, index=0, chars=13, level=0, gravity=south, flags=4, font=OMITTED, script=latin, language=en-us, 'some|bla|bla|'
i=2, index=13, no run, line end
i=3, index=13, chars=3, level=0, gravity=south, flags=0, font=OMITTED, script=latin, language=en-us, 'bla'
i=4, index=16, no run, line end
i=5, index=17, no run, line end
